### DISCUSSION TITLE
Invoice Issue with saved cards

### DISCUSSION THREAD

**Discussion by PrinzKenny1** [flutter-stripe/flutter_stripe#542] [Q&A]
Hey,
let's say I have a customer with a Visa Card and it's his default one. If he now creates an invoice and I get the PaymentIntent from it I can use the client secret, but the presentPaymentSheet Method will not work, because the PaymentIntent is in the requires_confirmation state, because the Visa Card automatically gets added as payment option.

Now my question: Is it possible to a) don't allow to automatically set it as payment option or b) make it possible that I can change the payment option with the presentPaymentSheet Method?

Or ist this Method only for PaymentIntent?

Greetings,
Calvin

**Comment by remonh87** [comment]
I am not sure if I understand you because the paymentsheet provides the payment options and you can choose the method you like based on what is specified on the payment intent. Requirement is that the paymentsheet allows the payment method (for example Ideal is supported)

**Reply by PrinzKenny1** [reply]
Yes, right. But I can't choose a payment option, if the intent is in the requires_confirmation state. At least I get an error saying:
`A PaymentIntent with status='requires_payment_method' or 'requires_action' is required.` And that my Payment Intent status is requires_confirmation (sorry I misspelled it in the main description of the problem).

**Reply by PrinzKenny1** [reply]
Here a step by step instruction to replicate it:
1. Create a user
2. Create an invoice for that user
3. The user pays it with e.g. credit card (the credit card automatically got saved as his standard payment method)
4. Create an invoice again (note that the invoice doesn't charge the card automatically because the collection method is set to send_invoice. Nevertheless does the card gets added as payment method for that invoice automatically because it's the standard payment method of the user.)

Since the invoice now has a payment method attached to it, it is in the requires_confirmation state. The payment option can still be changed at this moment but if I try to call the presentPaymentsheet Method I get the error above..

**Reply by sleewok** [reply]
Are you saving the default payment method for off_session use? If that is the case, try changing it to PaymentIntentsFutureUsage.OnSession if you are not charging them via subscription, etc.

**Reply by PrinzKenny1** [reply]
No, I'm not saving them for off session use, but they still get attached to the user. That's the weird behavior. I'm not really sure if it's a problem with stripe itself, because it only happens, if I pay with the provided link in the email, that was sent by stripe. 
But it shouldn't be a problem to change the payment method with the paymentSheet, should it? 

Noteable is that I'm currently not using invoices anymore and thus don't have the issue (because the user doesn't get an email with a link to pay). Nevertheless I think this topic needs some discussion for maybe other people with the same issue?

**Reply by sleewok** [reply]
Okay, that was my only lead. I wonder if there is some sort of configuration for the Stripe email that is sent. 

For future readers of this maybe this will help...From my understanding a Stripe API call needs to be made using PaymentIntent to update / confirm the payment_method that is attached to the intent (https://stripe.com/docs/api/payment_intents/confirm).

Glad you have a working solution now.
